bitnami/etcd/3.5/debian-12/rootfs/opt/bitnami/scripts/etcd/preupgrade.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+# Copyright Broadcom, Inc. All Rights Reserved.
+# SPDX-License-Identifier: APACHE-2.0
+
+# shellcheck disable=SC1091
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+# Load libraries
+. /opt/bitnami/scripts/libfs.sh
+. /opt/bitnami/scripts/libos.sh
+. /opt/bitnami/scripts/libetcd.sh
+
+# Load etcd environment settings
+. /opt/bitnami/scripts/etcd-env.sh
+
+########################
+# Return a comma separated list of <host>:<port> for each endpoint
+# based on "initial-cluster" flag value
+# ref: https://etcd.io/docs/latest/op-guide/clustering/#static
+# Globals:
+#   ETCD_INITIAL_CLUSTER 
+# Arguments:
+#   None
+# Returns:
+#   String
+########################
+endpoints_as_host_port() {
+    echo $ETCD_INITIAL_CLUSTER | tr -s ',' '\n' | awk -F '//' '{print $2}' | tr -s '\n' ',' | sed 's/,$//'
+}
+
+# Remove members that are not listed in ETCD_INITIAL_CLUSTER
+# from the cluster before running Helm upgrades that potentially scale
+# down the etcd cluster
+
+read -r -a extra_flags <<<"$(etcdctl_auth_flags)"
+is_boolean_yes "$ETCD_ON_K8S" && extra_flags+=("--endpoints=$(endpoints_as_host_port)")
+debug "Listing members"
+if ! current="$(etcdctl member list ${extra_flags[@]} --write-out simple | awk -F ", " '{print $3 ":" $1}')"; then
+    error "Unable to list members, are all members healthy?"
+    exit 1
+fi
+info "Current cluster members are: $(echo "$current" | awk -F: '{print $1}' | tr -s '\n' ',' | sed 's/,$//g')"
+
+expected="$(echo $ETCD_INITIAL_CLUSTER | tr -s ',' '\n' | awk -F= '{print $1}')"
+info "Expected cluster members are: $(echo "$expected" | tr -s '\n' ',' | sed 's/,$//g')"
+read -r -a obsolete_members <<<"$(comm -23 <(echo "$current" | awk -F: '{print $1}' | sort) <(echo "$expected" | sort))"
```

Comment:
small fix:
```suggestion
read -r -a obsolete_members <<<"$(comm -23 <(echo "$current" | awk -F: '{print $1}' | sort) <(echo "$expected" | sort) | tr -s '\n' ' ')"
```

Reply:
This change seems to break the script for me.

Reply:
Actually, something else is breaking. Let me investigate more.

Reply:
What problems are you experiencing? I works for me

Without this change only one member is removed given the `obsolete_members` wasn't an array but a string with the 1st obsolete member to remove, so when I did a test scaling down from 5 to 3 replicas it only removed one.

Reply:
Sorry, your suggestion works. I just had to fix the loop as well. Everything should work now.